RyuSocks/Packets/UaPPacket.cs

```diff
@@ -0,0 +1,21 @@
+namespace RyuSocks.Packets
+{
+    public struct UaPPacket(string username, string password) : IAuthenticationPacket
```

Comment:
This should be a class and renamed to something like `UsernameAndPasswordRequest`, since the RFC also calls it that.

I just added a commit to disable the ReSharper rule that suggested using primary constructors. We don't use them here.